sdk/metric/config.go

```diff
@@ -14,12 +14,123 @@
 
 package metric // import "go.opentelemetry.io/otel/sdk/metric"
 
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/otel/sdk/metric/view"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
 // config contains configuration options for a MeterProvider.
-type config struct{}
+type config struct {
+	res     *resource.Resource
+	readers map[Reader][]view.Config
+}
+
+// readerSignals returns a force-flush and shutdown function for a
+// MeterProvider to call in their respective options. All Readers c contains
+// will have their force-flush and shutdown methods unified into returned
+// single functions.
+func (c config) readerSignals() (forceFlush, shutdown func(context.Context) error) {
+	var fFuncs, sFuncs []func(context.Context) error
+	for r := range c.readers {
+		sFuncs = append(sFuncs, r.Shutdown)
+		fFuncs = append(fFuncs, r.ForceFlush)
+	}
+
+	return unify(fFuncs), unifyShutdown(sFuncs)
+}
+
+// unify unifies calling all of funcs into a single function call. It will
+// return nil if funcs is empty. All errors returned from calls to funcs will
+// be unify into a single error return value.
+func unify(funcs []func(context.Context) error) func(context.Context) error {
+	if len(funcs) == 0 {
+		return func(context.Context) error { return nil }
+	}
```

Comment:
nit: is this required? Seems like the fuction below handles the zero case correctly.

Reply:
It does. This was an attempt for the CPU to not have to go through a few extra cycles. Probably not worth it given the compiler probably optimizes it anyway, and if not it is not called in the hot-path. I'll remove it.